packages/graphql/src/translate/projection/elements/create-projection-subquery.ts

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { Node } from "../../../classes";
+import type { Context, GraphQLOptionsArg, GraphQLSortArg, GraphQLWhereArg, RelationField } from "../../../types";
+import * as CypherBuilder from "../../cypher-builder/CypherBuilder";
+import { createCypherWherePredicate } from "../../where/create-cypher-where-predicate";
+import type { RelationshipDirection } from "../../../utils/get-relationship-direction";
+import { createAuthPredicates } from "../../create-auth-and-params";
+import { AUTH_FORBIDDEN_ERROR } from "../../../constants";
+import { toNumber } from "../../../utils/utils";
+
+export function createProjectionSubquery({
+    parentNode,
+    whereInput,
+    node,
+    context,
+    alias,
+    nestedProjection,
+    nestedSubqueries,
+    relationField,
+    relationshipDirection,
+    optionsInput,
+    authValidateStrs,
+}: {
+    parentNode: CypherBuilder.Node;
+    whereInput?: GraphQLWhereArg;
+    node: Node;
+    context: Context;
+    alias: string; // TODO: this should be output variable instead
+    nestedProjection: string;
+    nestedSubqueries: CypherBuilder.Clause[];
+    relationField: RelationField;
+    relationshipDirection: RelationshipDirection;
+    optionsInput: GraphQLOptionsArg;
+    authValidateStrs: string[] | undefined;
+}): {
+    subquery: CypherBuilder.Clause;
+    projectionColumn: CypherBuilder.ProjectionColumn[];
```

Comment:
I've maybe just not seen this before, but "projection column" seems like a very odd name to me!

Reply:
It is the best name I could think for this entity based on [Cypher docs](https://neo4j.com/docs/cypher-manual/current/clauses/return/#return-column-alias)